core/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -53,7 +53,10 @@ private[spark] abstract class YarnSchedulerBackend(
    * This includes executors already pending or running.
    */
   override def doRequestTotalExecutors(requestedTotal: Int): Boolean = {
-    yarnSchedulerEndpoint.askWithRetry[Boolean](RequestExecutors(requestedTotal))
+    if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
```

Comment:
Can you explain how this method is called with dynamic allocation disabled?

The only call chain I can find starts with `ExecutorAllocationManager`, which is not instantiated when dynamic allocation is off.